File: week01/MINSANG/1859.java
import java.util.Scanner;

public class Solution {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        int T = sc.nextInt();
        
        for (int tc = 1; tc <= T; tc++) {
            int N = sc.nextInt();
            int[] prices = new int[N];
            
            for (int i = 0; i < N; i++) {
                prices[i] = sc.nextInt();
            }
            
            long totalProfit = 0; 
            int maxPrice = 0;
            
            for (int i = N - 1; i >= 0; i--) {
                int currentPrice = prices[i];
                
                if (currentPrice > maxPrice) {
                    maxPrice = currentPrice; 
                } else {
                    totalProfit += (maxPrice - currentPrice); 
                }
            }
            
            System.out.println("#" + tc + " " + totalProfit);
        }
        
        sc.close();
    }
}

File: week01/MINSANG/1859.py
T = int(input())
for tc in range(1,1+T):
    N = int(input())
    prices = list(map(int, input().split()))
    
    max_price = 0
    total_profit = 0
    
    for price in reversed(prices):
        if price > max_price:
            max_price = price
        else:
            total_profit += (max_price-price)
        
    print(f"#{tc} {total_profit}")
        